src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -0,0 +1,262 @@
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Textarea } from "@/components/ui/textarea";
+import { MembersProps } from "@/types";
+import { LeaveTypes } from "@/constant/constant";
+import { MemberSelector } from "@/common/MemberSelector/MemberSelector";
+
+export const leaveRequestSchema = z
+  .object({
+    memberId: z.string().min(1, {
+      message: "Member must be selected.",
+    }),
+    leaveType: z.string().min(1, {
+      message: "Leave Type must be selected.",
+    }),
+    startDate: z
+      .date({
+        message: "Start Date must be selected.",
+      })
+      .refine((date) => date >= new Date(), {
+        message:
+          "Start date must be greater than or equal to the current date.",
+      }),
+    endDate: z
+      .date({
+        message: "End Date must be selected.",
+      })
+      .refine((date) => date >= new Date(), {
+        message: "End date must be greater than or equal to the current date.",
+      }),
+    reasonForLeave: z.string().min(10, {
+      message: "Please provide a reason for leave.",
+    }),
+    status: z.string().min(1, {
+      message: "Status must be selected.",
+    }),
+  })
+  .refine((data) => data.endDate > data.startDate, {
+    message: "End date cannot be earlier than start date.",
+    path: ["endDate"],
+  });
+
+export const LeaveRequestFormBuilder = ({
+  members,
+}: {
+  members: MembersProps;
+}) => {
+  const { data } = members;
+  const form = useForm<z.infer<typeof leaveRequestSchema>>({
+    resolver: zodResolver(leaveRequestSchema),
+    defaultValues: {
+      memberId: "",
+      leaveType: "",
+      startDate: undefined,
+      endDate: undefined,
+      reasonForLeave: "",
+      status: "Pending",
+    },
+    mode: "all",
+  });
+
+  const { errors, isValid } = form.formState;
+
+  function onSubmit(values: z.infer<typeof leaveRequestSchema>) {
+    console.log(values);
+  }
+  return (
+    <FormWrapper>
+      <h1 className="text-lg font-semibold text-gray-800 my-4">
+        Request Leave
+      </h1>
+      <Form {...form}>
+        <form
+          data-testid="form"
+          onSubmit={form.handleSubmit(onSubmit)}
+          className="space-y-4"
+        >
+          <MemberSelector control={form.control} name="memberId" />
+
+          <FormField
+            control={form.control}
+            name="leaveType"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Leave Type</FormLabel>
+                <FormControl>
+                  <Select
+                    {...field}
+                    value={field.value}
+                    onValueChange={field.onChange}
+                  >
+                    <SelectTrigger data-testid="leave-type" id="leaveType">
+                      <SelectValue placeholder="Select a leave type" />
+                    </SelectTrigger>
+                    <SelectContent>
+                      <SelectItem value={LeaveTypes.Personal}>
```

Comment:
can we use here map method for it

Reply:
No, because there are only three items, so I haven't used the map() method. Additionally, the map() method allocates memory.